task2.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Notification
+  def self.included(base)
+    base.extend(ClassMethods)
+  end
+
+  module ClassMethods
+    def log
+      puts File.read("#{name.downcase}.log")
+    end
+  end
+
+  def send_message(recepient)
+    puts "Sending #{self.class.name} to #{recepient}"
+  end
+
+  def add_to_log(recepient)
+    File.open("./#{self.class.name.downcase}.log", 'a') do |c|
+      c.puts("#{recepient}: got an error \n ")
+    end
+  end
+end
+
+class Sms
+  include Notification
+  VALID_NUMBER = /\A(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]?\d{3}[\s.-]?\d{4}\z/.freeze
```

Comment:
classes Sms and Email should contain only `include Notification` all other code needs to be inside this module